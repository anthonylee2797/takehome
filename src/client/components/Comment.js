import React from 'react'
import moment from 'moment'
import PropTypes from 'prop-types'

/* eslint-disable react/display-name */

const Comment = React.memo((props) => {
  const { name, message, created } = props.comment
  const date = moment.utc(created).fromNow()

  return (
    <div className="comment">
      <span className="comment-top">
        <span className='comment-top-container'>{name}</span>
        {message}
      </span>
      <p className="comment-bottom" >{date}</p>
    </div>
  )
})

Comment.propTypes = {
  comment: PropTypes.shape({
    message: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    created: PropTypes.string.isRequired
  })
}

export default Comment
