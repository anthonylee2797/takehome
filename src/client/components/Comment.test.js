import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Comment from './Comment'
import moment from 'moment'


test('it should render comments with props', () => {
  jest.spyOn(moment, 'utc').mockImplementation(() => {
    return {
      fromNow: () => 'a few seconds ago'
    }
  })

  const testProps = {name: 'Mailchimp', message: 'I like emails', created: ""}
  const testComment = render(<Comment comment={testProps}/>)

  expect(testComment.container).toMatchSnapshot()
})
