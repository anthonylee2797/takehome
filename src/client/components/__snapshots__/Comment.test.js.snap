// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render comments with props 1`] = `
<div>
  <div
    class="comment"
  >
    <span
      class="comment-top"
    >
      <span
        class="test"
      >
        Mailchimp
      </span>
      I like emails
    </span>
    <p
      class="comment-bottom"
    >
      a few seconds ago
    </p>
  </div>
</div>
`;
